### PR TITLE
Getting Started with Time Series Decomposition in python 

### LINKED ISSUE #5534: [Machine Learning] Getting Started with Time Series Decomposition in python
*NOTE*: All **1st-time contributors** should know Topic approval and PR submission does NOT guarantee your Topic/Article will be published. Our team of Peer Reviewers and Content Moderators will review all PRs that come in to make sure they adhere to the standard of quality we expect from the EngEd community. 

We expect all community memebers to go through provided examples, resources, and previously published material before submitting content. As a rule of thumb, please only submit articles (pull requests) that are complete, formatted correctly, and include a fully polished article (ready to be published and error free). 

All revisions and edits should be *completed on your own forked repo (as to not take up room in the queue)*. **Any PR that is submitted incomplete (meaning not ready to be published as is) will be closed.** A PR being closed does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

### Topic Suggestion
Enter topic suggestion here...

#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximise your chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published (subject to vary depending on _volume and/or backlog of articles_).

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Writing sample(s): 
Include any links or writing samples - to help our team better gauge your writing quality.

### Introduction
Time Series Decomposition is one of the foundational topics of time series modeling. It involves deconstructing a time series into its different components: seasonality, trend, and the leftover residual. We do this because the model performs much better when we remove the already known source of variation. Since the trend and seasonality are deterministic, they can be decomposed from the time series such that we remain the indeterministic component that is not possible to determine(residual). From this, we can now learn the autocorrelation structure of the time series using this indeterministic structure of residuals.

This article will discuss the decomposition of the time series data into these components and the type of models we use when decomposing this data(Additive, Multiplicative, and Pseudo-additive Models). 

Before decomposing our time series, we shall first have to discuss in detail what these components are:

1. Trend: This is the general long-term direction of the time series, which can either be increasing, decreasing, or constant.

1. Seasonality: The periodic behavior of the time series that occurs within a period of less than one year.

1. Residual: The irregular fluctuations that can't be predicted using the trend or the seasonality.

### Learning outcomes of this article
What will we accomplish;
1. First, we'll separately construct each time series component. 
2. Then, we'll bring the three components together to create a simulated time series dataset. 
3. Finally, we'll leverage time series model decomposition in python to deconstruct the series. 

We'll complete this process for the two most common methods: additive and multiplicative models used in time series.

The whole point of this article will be to get the learner familiar with time-series data and simple modeling approaches. 

#### Key takeaways 
An attentive learner should walk away with:
1. A practical understanding of the three components of time series data 
2. A better understanding of additive and multiplicative models
3. A simple way to decompose time series data in python

#### Article quality
Time series is a complicated topic that, if not well covered it can pose a great challenge to the learner. With the keen observation of other topics on the time series decomposition, I have noticed that many authors take advantage of libraries when it comes to implementing. As a result, the learner can't relate what they theoretically learn and what they implement. This article will teach how to decompose time series from scratch using pure programming skills in python and later leverage the existing models for implementation.

Also, many topics on the time series decomposition use datasets that need to pass through a series of time series data preprocessing. Although data preprocessing is an essential activity, it should not be included in topics meant to get learners started. An absolute learner needs to be provided with only the content that is direct to the point. To not load learners with 'insignificant' content, this article will generate a dataset that is ready to use. So, to start with, we shall generate data for each component and combine them to form a time series that later we shall decompose buck to its initial components. This article will engage the learners' attention to a higher degree.

#### References
N/A

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by sumbaelvis**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5534
> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @sumbaelvis 👋 .
My name is Briana and I will be reviewing your article.
First of all, good job on the article. It is a very interesting read and very minimal corrections to be made.
Just a few changes I need you to make that I have mentioned as we continue with the tech review.

In case of any question or help needed please do not hesitate to ping me here :smile:.
Also, notify me when you are done.

Regards.

File: content/articles/time-series-decomposition-in-python/index.md [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+  - url: /time-series-decomposition-in-python/hero.png
+    alt: Getting started with Time Series Decomposition in Python Hero Image
+---
+### Time Series Decomposition in Python
+Time series refers to the data that is sequentially collected over time. Various components usually contribute to any observed time series value at any particular point in time.
+<!--more-->
+Thus a time series can be decomposed such that each component exists independently. Usually, it is challenging to work with a model whose most of the cause of variation is unknown. In time series, some of the causes of variation to the model are deterministic.
+
+Analyzing a model with many causes of variation is computationally expensive. Therefore, we decompose those components whose variation is deterministic. The indeterministic ones, residuals, are then used to analyze of autocorrelation.
+
```
Comment (BrianaNzivu) [inline content/articles/time-series-decomposition-in-python/index.md:23]: Let us also have the table of contents of this article.

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]

**Comment by BrianaNzivu** [comment]
Tech review done

**Comment by WanjaMIKE** [comment]
Plagiarism check is done

**Review by BrianaNzivu — APPROVED** [review]
Hey @sumbaelvis  👋 .

Very good job on the article. It is a very interesting read and has minimal mistakes 👏 .

Just a few things I'd like to point out:
- I suggest that you check out the [structure](https://github.com/section-engineering-education/engineering-education/tree/master/new_contributors) for the various articles accepted by Section.
- I suggest that you check out our resources page which has links to various resourceful tools for writing just to minimize the changes being made by the reviewer.
- Lastly please go through ALL the corrections I have made so that you can not repeat the same mistakes.

That's it! :smile: 

A final review will be done now by a content moderator.
Otherwise, congratulations on publishing your article🍾.

In case of any question or clarification (currently or in the future), do not hesitate to reach out via [Slack](https://app.slack.com/client/T5X14SH7G/D01A6MT673R) or my personal email (briananzivu@gmail.com). 
I am **ALWAYS** here to help.

It has been a pleasure reading and reviewing this article.

Looking forward to seeing you contribute more to EngEd. 🚀

Once again good job! 👍 

Kind Regards
*P.s Happy new year and a prosperous 2022*

**Comment by ahmadmardeni1** [comment]
Great work. Please go over the changes I've made. @BrianaNzivu @sumbaelvis

**Pull request merged**